src/components/modules/banners/contributionsTemplate/ContributionsTemplateTicker.tsx

```diff
@@ -0,0 +1,197 @@
+import React, { useEffect, useState } from 'react';
+import { css, SerializedStyles } from '@emotion/core';
+import { palette } from '@guardian/src-foundations';
+import { headline } from '@guardian/src-foundations/typography';
+import { useHasBeenSeen, HasBeenSeen } from '../../../../hooks/useHasBeenSeen';
+import { TickerSettings } from '../../../../lib/variants';
+
+// This ticker component provides an animated progress bar and counter for the
```

Comment:
is this very different from the epic ticker?

Reply:
it's really quite similar.
```diff
2c2
< import { css } from '@emotion/core';
---
> import { css, SerializedStyles } from '@emotion/core';
4d3
< import { from } from '@guardian/src-foundations/mq';
6,7c5,6
< import { useHasBeenSeen, HasBeenSeen } from '../../../hooks/useHasBeenSeen';
< import { TickerSettings } from '../../../lib/variants';
---
> import { useHasBeenSeen, HasBeenSeen } from '../../../../hooks/useHasBeenSeen';
> import { TickerSettings } from '../../../../lib/variants';
17d15
<     margin-bottom: 15px;
21c19
< const totalCountStyles = css`
---
> const totalCountStyles = (colour: string): SerializedStyles => css`
22a21
>     color: ${colour};
25c24
< const soFarCountStyles = css`
---
> const soFarCountStyles = (colour: string): SerializedStyles => css`
26a26
>     color: ${colour};
30c30
<     ${headline.xxxsmall({ fontStyle: 'italic' })};
---
>     ${headline.xxxsmall()};
33c33
< const progressBarHeight = 10;
---
> const progressBarHeight = 12;
46a47
>     background: white;
69c70,75
< const filledProgressStyles = (end: number, runningTotal: number, total: number) =>
---
> const filledProgressStyles = (
>     end: number,
>     runningTotal: number,
>     total: number,
>     colour: string,
> ): SerializedStyles =>
78c84
<         background-color: ${palette.brandAlt.main};
---
>         background-color: ${colour};
81,88c87
< const goalReachedGoalContainerStyles = css`
<     display: none;
<     ${from.tablet} {
<         display: initial;
<     }
< `;
< 
< const goalContainerStyles = (goalReached: boolean) => css`
---
> const goalContainerStyles = css`
93d91
<     ${goalReached && goalReachedGoalContainerStyles}
96c94
< const goalMarkerStyles = (transform: string) => css`
---
> const goalMarkerStyles = (transform: string): SerializedStyles => css`
123,124c121
<     total: number;
<     goal: number;
---
>     accentColour: string;
127c124
< export const ContributionsEpicTicker: React.FC<Props> = ({ settings, total, goal }: Props) => {
---
> const ContributionsEpicTicker: React.FC<Props> = ({ settings, accentColour }: Props) => {
130a128,130
>     const total = settings.tickerData?.total || 0;
>     const goal = settings.tickerData?.goal || 0;
> 
162d161
<     const goalReached = total >= goal;
172,180c171,173
<                     <div css={soFarCountStyles}>
<                         {goalReached
<                             ? settings.copy.goalReachedPrimary
<                             : `${currencySymbol}${runningTotal.toLocaleString()}`}
<                     </div>
<                     <div css={countLabelStyles}>
<                         {goalReached
<                             ? settings.copy.goalReachedSecondary
<                             : settings.copy.countLabel}
---
>                     <div css={soFarCountStyles(accentColour)}>
>                         {currencySymbol}
>                         {runningTotal.toLocaleString()}
181a175
>                     <div css={countLabelStyles}>contributed</div>
184,191c178,181
<                 <div css={goalContainerStyles(goalReached)}>
<                     <div css={totalCountStyles}>
<                         {goalReached
<                             ? `${currencySymbol}${total.toLocaleString()}`
<                             : `${currencySymbol}${goal.toLocaleString()}`}
<                     </div>
<                     <div css={countLabelStyles}>
<                         {goalReached ? settings.copy.countLabel : 'our goal'}
---
>                 <div css={goalContainerStyles}>
>                     <div css={totalCountStyles(accentColour)}>
>                         {currencySymbol}
>                         {goal.toLocaleString()}
192a183
>                     <div css={countLabelStyles}>our goal</div>
198c189
<                     <div css={filledProgressStyles(end, runningTotal, total)}></div>
---
>                     <div css={filledProgressStyles(end, runningTotal, total, accentColour)}></div>
204a196,197
> 
> export default ContributionsEpicTicker;
```

Here's a diff. There are some styling differences e.g. italic vs non italic font, accent colour, margins. And some logic has been simplified in this one. In fact, It think I could remove even more logic e.g. the animation can start right away, it doesn't need to wait for it to be scrolled into view

Reply:
I think we want to keep the ability to configure the label copy, in case we run another campaign where we're counting supporters rather than money. It'll get confusing if the epic and banner tickers have different functionality

Reply:
Yeah makes sense. My general approach was to make the components as simple as possible with the idea that we would start to add functionality as and when we need it. This is quite core though, so happy to add that if you want